### DISCUSSION TITLE
Determining if the CSLA is running in "client" or a "server" mode

### DISCUSSION THREAD

**Discussion by GreatBarrier86** [MarimerLLC/csla#1734] [Questions] [answered]
Question. Hope this should be relatively easy. Is there a built-in way to determine if CSLA is running in client/server mode? I would imagine a few possibilities, but i was curious first if there is a built-in method. My goal is to only execute certain code in OnDeserialized if we're in client mode, but i dont know if there is a good method.

The problem with my ideas below is that if you have a 4-tier application (client -> another server -> app server  -> database), both of the ideas would show that it's Client mode on **another server** because those values would be set.

```
!String.IsNullOrWhitespace(Csla.ApplicationContext.DataPortalUrlString)
or
Csla.ApplicationContext.DataPortalProxy != "local"
```
My last idea would be just a bool we set somewhere that is `IsClientMode` that we set in each of our application's inits.

**Comment by JacoJordaan** [comment, marked as answer]
You can use `Csla.ApplicationContext.ExecutionLocation` to determine where the code is currently running. 

E.g. check to see if running on client:
````
if (Csla.ApplicationContext.ExecutionLocation != ApplicationContext.ExecutionLocations.Server)
{

}
''''

**Reply by JacoJordaan** [reply]
Sorry, I may have misunderstood your question...

**Reply by GreatBarrier86** [reply]
It seems like you did. The ExecutionLocation values are correct as i expected them to be when i debugging both the client and server. Unless i misunderstood the use case for that property.
